docs/API.md

```diff
@@ -2376,3 +2398,91 @@ Example: `GET changes/published?limit=10&start=31-07-2014`
     }
 ]
 ```
+
+### Uploads
+
+When uploading a large resource to a charm, it can be unreliable
+to do it in a single HTTP request, so the charm store provides a
+way to upload the data as a set of parts that are then stitched
+together to make the whole resource.
+
+#### POST /upload[?expires=*expires*]
+
+This endpoint starts an upload. If the uploaded data
+is not used within the given expiry duration, it
+will be discarded. The response holds a JSON
+object containing information about the upload:
+
+```go
+type NewUploadResponse struct {
+	// UploadId holds the id of the upload.
+	UploadId string
+
+	// Expires holds when the upload id expires.
+	Expires time.Time
```

Comment:
As discussed on IRC perhaps it is worth mentioning that the user-supplied expiration time must be <= 24h from the upload start time.

Are there restrictions on the format or timezone used? Does it have to be Zulu?